vm/tests/vm.rs

```diff
@@ -255,6 +258,60 @@ fn bls_multisig_signature() {
     assert!(!valid, "Multisig Signature verification expected to fail");
 }
 
+#[test]
+fn keccak256() {
+    let vm = VM::ephemeral().expect("Instantiating VM should succeed");
+    let (mut session, contract_id) = instantiate(&vm, 0);
+
+    let input = hex::decode("ab00ef77dd33ee006d8a5c9b62")
+        .expect("Hex decoding should work");
+
+    let output = session
+        .call::<_, [u8; 32]>(contract_id, "keccak256", &input, POINT_LIMIT)
+        .expect("Querying should succeed")
+        .data;
+
+    assert_eq!(
+        "48299ecd7ccb5655d3be5747703c44137173e1c5ef2ec9e175bffe9e2c5e3eda",
+        hex::encode(output),
+    );
+}
+
+#[test]
+fn verify_secp256k1() {
+    let vm = VM::ephemeral().expect("Instantiating VM should succeed");
+    let (mut session, contract_id) = instantiate(&vm, 0);
+
+    let signing_key = SigningKey::random(&mut OsRng);
+    let verifying_key = signing_key.verifying_key();
+    let message = b"test message".to_vec();
+    let message_hash = Keccak256::digest(message).to_vec();
+    let signature: Signature = signing_key.sign(&message_hash);
+    let mut arg = (
+        message_hash,
+        verifying_key.to_sec1_bytes().to_vec(),
+        signature.to_bytes().to_vec(),
+    );
+
+    let verification_result = session
+        .call::<_, bool>(contract_id, "verify_secp256k1", &arg, POINT_LIMIT)
+        .expect("Querying should succeed")
+        .data;
+
+    assert!(verification_result);
+
+    let bad_message = b"bad test message".to_vec();
+    let bad_message_hash = Keccak256::digest(bad_message).to_vec();
+    arg.0 = bad_message_hash;
+
+    let verification_result = session
+        .call::<_, bool>(contract_id, "verify_secp256k1", &arg, POINT_LIMIT)
+        .expect("Querying should succeed")
+        .data;
+
+    assert!(!verification_result);
```

Comment:
maybe also add a test where the verification fails due to a different key being used

Reply:
done